tests/test_blockUtils.py

```diff
@@ -0,0 +1,198 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Test cases for utils."""
+
+import os
+import unittest
+import lsst.utils.tests
+import pandas as pd
+import numpy as np
+import asyncio
+import json
+
+from lsst.utils import getPackageDir
+from lsst.summit.utils.efdUtils import makeEfdClient
+from lsst.summit.utils.blockUtils import (
+    BlockParser,
+)
+
+__all__ = (
+    "writeNewBlockInfoTestTruthValues",
+)
+
+DELIMITER = "||"  # don't use a comma, as str(list) will naturally contain commas
+
+
+def getBlockInfoTestTruthValues():
+    """Get the current truth values for the block information.
+
+    Returns
+    -------
+    data : `dict` [`tuple` [`int`, `int`], `str`]
+        The block info truth data.
+    """
+    packageDir = getPackageDir("summit_utils")
+    dataFilename = os.path.join(packageDir, "tests", "data", "blockInfoData.json")
+
+    with open(dataFilename, 'r') as f:
+        loaded = json.loads(f.read())
+
+    data = {}
+    for dayObsSeqNumStr, line in loaded.items():
+        dayObs = int(dayObsSeqNumStr.split(f'{DELIMITER}')[0])
+        seqNum = int(dayObsSeqNumStr.split(f'{DELIMITER}')[1])
+        data[dayObs, seqNum] = line
+    return data
+
+
+def writeNewBlockInfoTestTruthValues():
+    """This function is used to write out the truth values for the test cases.
+
+    If bugs are found in the parsing, it's possible these values could change,
+    and would need to be updated. If that happens, run this function, and check
+    the new values into git.
+    """
+    dayObs = 20230615
+    blockParser = BlockParser(dayObs)
+
+    data = {}
+    for block in blockParser.getBlockNums():
+        seqNums = blockParser.getSeqNums(block)
+        for seqNum in seqNums:
+            blockInfo = blockParser.getBlockInfo(block, seqNum)
+            line = (
+                f"{blockInfo.blockId}{DELIMITER}"
+                f"{blockInfo.begin}{DELIMITER}"
+                f"{blockInfo.end}{DELIMITER}"
+                f"{blockInfo.salIndices}{DELIMITER}"
+                f"{blockInfo.tickets}{DELIMITER}"
+                f"{len(blockInfo.states)}"
+            )
+            # must store as string not tuple for json serialization
+            data[f'{block}{DELIMITER}{seqNum}'] = line
+
+    packageDir = getPackageDir("summit_utils")
+    dataFilename = os.path.join(packageDir, "tests", "data", "blockInfoData.json")
+    with open(dataFilename, 'w') as f:
+        json.dump(data, f)
+
+
+class BlockParserTestCase(lsst.utils.tests.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        try:
+            cls.client = makeEfdClient()
+        except RuntimeError:
```

Comment:
Please add a comment describing when you would expect `makeEfdClient` to fail.

Reply:
That's a really good point, and I actually don't know when I would expect that. Maybe when running from some place that isn't one of our systems? But I think the more important point that you've made me realise here, is that I've not annotated this to use the `vcr` package here, so this will end up contacting an actual EFD, which is no good at all! I will get that part fixed up right now!